source/agora/consensus/protocol/Nominator.d

```diff
@@ -123,7 +128,8 @@ extern(D):
         this.scp = createSCP(this, node_id, IsValidator, no_quorum);
         this.taskman = taskman;
         this.ledger = ledger;
-        this.restoreSCPState(ledger);
+        this.restoreSCPState();
+        this.latest_envelopes = this.getLatestEnvelopes(":memory:");
```

Comment:
I'll give you a hint: You are restoring SCP state, and then retrieving the latest envelopes after.